NitroxClient/Debuggers/Drawer/NitroxGUILayout.cs

```diff
@@ -0,0 +1,194 @@
+                activeConvertibleField = -1;
+                if (parsed)
+                {
+                    break;
+                }
+
+                value = TryParseIConvertible(value, strValue, out IConvertible newValue) ? newValue : activeConvertibleFieldLastValue;
+                break;
+            }
+        }
+
+        return value;
+    }
+
+    private static bool TryParseIConvertible(IConvertible type, string inputString, out IConvertible newValue)
+    {
+        bool parsed;
+        switch (type)
+        {
+            case short:
+                parsed = short.TryParse(inputString, NumberStyles.Integer, CultureInfo.CurrentCulture, out short newShort);
+                newValue = newShort;
+                break;
+            case ushort:
+                parsed = ushort.TryParse(inputString, NumberStyles.Integer, CultureInfo.CurrentCulture, out ushort newUShort);
+                newValue = newUShort;
+                break;
+            case int:
+                parsed = int.TryParse(inputString, NumberStyles.Integer, CultureInfo.CurrentCulture, out int newInt);
+                newValue = newInt;
+                break;
+            case uint _:
+                parsed = uint.TryParse(inputString, NumberStyles.Integer, CultureInfo.CurrentCulture, out uint newUInt);
+                newValue = newUInt;
+                break;
+            case long:
+                parsed = long.TryParse(inputString, NumberStyles.Integer, CultureInfo.CurrentCulture, out long newLong);
+                newValue = newLong;
+                break;
+            case ulong:
+                parsed = ulong.TryParse(inputString, NumberStyles.Integer, CultureInfo.CurrentCulture, out ulong newULong);
+                newValue = newULong;
+                break;
+            case float:
+                parsed = float.TryParse(inputString, NumberStyles.Float, CultureInfo.CurrentCulture, out float newFloat);
+                newValue = newFloat;
+                break;
+            case double:
+                parsed = double.TryParse(inputString, NumberStyles.Float, CultureInfo.CurrentCulture, out double newDouble);
+                newValue = newDouble;
+                break;
+            default:
+                parsed = false;
+                newValue = null;
+                break;
+        }
+
+        return parsed;
+    }
+
+    public static int SliderField(int value, int minValue, int maxValue, float valueWidth = VALUE_WIDTH) => (int)SliderField((float)value, minValue, maxValue, valueWidth);
+
+    public static float SliderField(float value, float minValue, float maxValue, float valueWidth = VALUE_WIDTH)
+    {
+        //TODO: Implement slider (if possible at all)
+        return Math.Max(minValue, Math.Min(maxValue, FloatField(value, valueWidth)));
+    }
+
+    public static T EnumPopup<T>(T selected, float buttonWidth = VALUE_WIDTH) where T : Enum
+    {
+        //TODO: Implement selecting of Enum in new window
```

Comment:
Not a blocker but some todo here were these intended to be implemented now or todo for future?